### PR TITLE
[BUGFIX] SQLAlchemy selectable Bug fix

### PR DESCRIPTION
**Pull request by tommy-watts-depop**
Change related to issue Redshift spectrum temp tables should be explicitly declared as text(<temp-table>) #6148

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- pass selectable to sa.text

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by anthonyburdi** [comment]
Thank you for the contribution @tommy-watts-depop! Taking a look at why the tests were failing on your initial commit 👀

**Comment by anthonyburdi** [comment]
I think we may have to only convert the query to text when we are using the redshift dialect. I'm testing out that change here through our full CI pipeline: https://github.com/great-expectations/great_expectations/pull/6150

File: great_expectations/expectations/metrics/util.py [hunk 1]
```diff
@@ -570,7 +570,9 @@ def column_reflection_fallback(
         if isinstance(selectable, TextClause):
             query: TextClause = selectable
         else:
-            query: Select = sa.select([sa.text("*")]).select_from(selectable).limit(1)
+            query: Select = (
+                sa.select([sa.text("*")]).select_from(sa.text(selectable)).limit(1)
+            )
```
Comment (anthonyburdi) [inline great_expectations/expectations/metrics/util.py]: Hi @tommy-watts-depop - thanks again for this contribution. I think we will want to limit the effect to just the Redshift dialect, so would you mind making this change? After that, tests should pass and we will be good to merge this PR.

```suggestion
            if dialect.name.lower() == GESqlDialect.REDSHIFT:
                # Redshift needs temp tables to be declared as text
                query: Select = (
                    sa.select([sa.text("*")]).select_from(sa.text(selectable)).limit(1)
                )
            else:
                query: Select = (
                    sa.select([sa.text("*")]).select_from(selectable).limit(1)
                )
```

Comment (anthonyburdi) [reply great_expectations/expectations/metrics/util.py]: You'll also need to add this import: `from great_expectations.execution_engine.sqlalchemy_dialect import GESqlDialect`

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Hi @tommy-watts-depop, thanks for the contribution! I have a minor suggestion but otherwise this looks good!

File: great_expectations/expectations/metrics/util.py (hunk 1 above)
Comment (tommy-watts-depop) [reply great_expectations/expectations/metrics/util.py]: Done

Comment (anthonyburdi) [reply great_expectations/expectations/metrics/util.py]: Wow - rapid response! I think you also need to commit the suggestion in addition to adding the import.

Comment (tommy-watts-depop) [reply great_expectations/expectations/metrics/util.py]: @anthonyburdi Apologies forgot to save the file ! repushed

File: great_expectations/expectations/metrics/util.py [hunk 2]
```diff
@@ -7,8 +7,10 @@
 from packaging import version
 
 from great_expectations.execution_engine.util import check_sql_engine_dialect
+from great_expectations.execution_engine.sqlalchemy_dialect import GESqlDialect
 from great_expectations.util import get_sqlalchemy_inspector
```
Comment (anthonyburdi) [inline great_expectations/expectations/metrics/util.py]: This should make isort happy:
```suggestion
from great_expectations.execution_engine.sqlalchemy_dialect import GESqlDialect
from great_expectations.execution_engine.util import check_sql_engine_dialect
from great_expectations.util import get_sqlalchemy_inspector
```

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Pull request merged**